src/main/scala/epic/logo/FeatureVector.scala

```diff
@@ -0,0 +1,50 @@
+package epic.logo
+
+import scala.collection.JavaConversions._
+import breeze.linalg._
+import breeze.math.MutableInnerProductModule
+
+case class FeatureVector[W](var data :W)(implicit space: MutableInnerProductModule[W, Double]) {
+  import space._
+
+  def normSquared() = data dot data
+
+  def ^(x : Int) = {
```

Comment:
I agree with the eyes.... this is incredibly problematic. Should not be here ! :(